### PR TITLE
fix(files): Correctly validate new node name

### LINKED ISSUE #45409: Nextcloud 28 and 29 : Properly pre-select new file and folder default name when creating them (for quick renaming)
**Renaming a new folder before creation :** 
- In Nextcloud 29, when we create a folder in “Files”, the folder name in the creation popup is not selected. This forces us to select the name, then proceed with renaming, and therefore affects the user experience.
- In Nextcloud 28, 27, the default name was selected, enabling on-the-fly renaming just before confirming (by pressing enter and clicking on the arrow).

**Renaming a new file before creation :** 
- In Nextcloud 28, when a file is created (for example, an odt file), the default file name and extension are selected, making it impossible to rename the file properly without first selecting the file name, without the extension. 
- In Nextcloud 29, nothing is pre-selected, so you have to select the file name with the mouse in order to proceed with renaming. 
- In Nextcloud 27, the file name (without extension) was selected, enabling on-the-fly renaming just before confirming (by pressing enter and clicking on the arrow).

### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are interested into the same feature.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments. 

**Is your feature request related to a problem? Please describe.**

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by susnux**
* Resolves https://github.com/nextcloud/server/issues/45409
* Visual feedback for #44963 

## Summary

This includes two fixes:
1. The name in the "new node" dialog is correctly selected (e.g. `file.txt` only `file` is selected by default), to allow quick naming
2. `@nextcloud/files` functions for filename validation are used, this allows to use new Nextcloud 30 capabilities (e.g. reserved names)

### Screen recording

[Bildschirmaufnahme_20240725_215605.webm](https://github.com/user-attachments/assets/a760a2b0-d972-4a36-9fa9-0186bc28adcc)

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [x] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- [x] Screenshots before/after for front-end changes
- [x] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required
- [x] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

### CODE REVIEW

File: apps/files/src/components/FileEntry/FileEntryName.vue [hunk 1]
```diff
@@ -123,7 +123,7 @@ export default defineComponent({
 
 		renameLabel() {
 			const matchLabel: Record<FileType, string> = {
-				[FileType.File]: t('files', 'File name'),
+				[FileType.File]: t('files', 'Filename'),
```
Comment (Fenn-CS) [inline apps/files/src/components/FileEntry/FileEntryName.vue:126]: Why?

File: apps/files/src/components/FileEntry/FileEntryName.vue [hunk 2]
```diff
@@ -27,7 +27,7 @@
 		class="files-list__row-name-link"
 		data-cy-files-list-row-name-link
 		v-bind="linkTo.params">
-		<!-- File name -->
+		<!-- Filename -->
```
Comment (Fenn-CS) [inline apps/files/src/components/FileEntry/FileEntryName.vue:30]: Why?

**Review by nfebe — APPROVED** [review]
Except for synonym addition, lgtm

File: cypress/e2e/files/files-renaming.cy.ts [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+	}))
+
+	it('can rename a file', () => {
+		// All are visible by default
+		getRowForFile('file.txt').should('be.visible')
+
+		triggerActionForFile('file.txt', 'rename')
+
+		getRowForFile('file.txt')
+			.findByRole('textbox', { name: 'Filename' })
```
Comment (skjnldsv) [inline cypress/e2e/files/files-renaming.cy.ts:34]: Fancy!

**Review by skjnldsv — APPROVED** [review]

**Comment by Jerome-Herbinet** [comment]
Hi,
It seems working for odt, ods, otp (Collabora) file types (except odg), but not for other file types : odg (Collabora), md (Nextcloud Text), and docx, xlsx, pptx (OnlyOffice). Tested on NC 30.0.0 beta 2.
Do you want me to create another issue, or will you create another PR on the way to fix this ?
Thanks

**Pull request merged**